.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,31 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: production
```

Comment:
We're using [GitHub environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) here so that only runs on `main` have access to the secrets they'll need for extension publication